hdc/algo/accessors.py

```diff
@@ -497,33 +496,31 @@ def spi(
 
         tix = self._obj.get_index("time")
 
-        calstart_ix = 0
-        if calibration_start is not None:
-            calstart = pd.Timestamp(calibration_start)
-            if calstart > tix[-1]:
-                raise ValueError(
-                    "Calibration start cannot be greater than last timestamp!"
-                )
-            (calstart_ix,) = tix.get_indexer([calstart], method="bfill")
+        if calibration_start is None:
+            calibration_start = tix[0]
 
-        calstop_ix = tix.size
-        if calibration_stop is not None:
-            calstop = pd.Timestamp(calibration_stop)
-            if calstop < tix[0]:
-                raise ValueError(
-                    "Calibration stop cannot be smaller than first timestamp!"
-                )
-            (calstop_ix,) = tix.get_indexer([calstop], method="ffill") + 1
+        if calibration_stop is None:
+            calibration_stop = tix[-1]
 
-        if calstart_ix >= calstop_ix:
-            raise ValueError("calibration_start < calibration_stop!")
+        if calibration_start > tix[-1:]:
```

Comment:
is this comparing `str > [str]`?

Reply:
comparing `str` to `pd.Timestamp`

Reply:
why `tix[-1:]` and not `tix[-1]`?

Reply:
needs to remain a `pd.DateTimeIndex` with one element to work ... maybe bit hacky & counterintuitive

Reply:
we could explicitly convert the string to a date object (`pd.Timestamp` or whatever) and then compare the elements if that's clearer